protocol/0053-PERP-product_builtin_perpetual_future.md

```diff
@@ -83,62 +83,16 @@ Every time a [mark to market settlement](./0003-MTMK-mark_to_market_settlement.m
 
 When the `settlement_schedule` event is received we need to calculate the funding payment. Store the current vega time as `funding_period_end`.
 
-If there are no oracle data points with a timestamp less than `funding_period_end` available then funding payment is skipped and `funding_period_start` gets overwritten with `funding_period_end`.
+Skip the funding payment calculation (set payment to `0`) if any of the following conditions is met:
 
-If such points are available then the calculations discussed in the following subsections get executed and funding payments get exchanged.
+- no spot (external) data has been ingested since market was created,
+- market has been in auction throughout the entire funding period.
 
-#### TWAP spot price calculation
+Please refer to the following subsections for the details of calculation of the funding payment if none of the above conditions are met.
 
-Traverse all the available oracle data point tuples `(s,t)` and calculate the time-weighted average spot price (`s_twap`) as:
+#### TWAP calculation
 
-```go
```

Comment:
are we really deleting the calculation formula / code?

Reply:
That was the intention as it will be fairly convoluted to explain in pseduocode now and was hoping the description + detailed AC example would be enough, but can add it if you think otherwise.

Reply:
@davidsiska-vega @EVODelavega I've [reinstated the pseudocode](https://github.com/vegaprotocol/specs/pull/2001/commits/68155efe0faa89df76906bf558e9cb212cac4121). If you could give it a once over when you have a moment and approve if you're happy that'd be amazing :pray: